src/picongpu/include/particles/bremsstrahlung/Bremsstrahlung.tpp

```diff
@@ -0,0 +1,252 @@
+    if(vecOrtho1Abs == float_X(0.0))
+        vecOrtho1_norm = float3_X(1.0, 0.0, 0.0);
+    else
+        vecOrtho1_norm = vecOrtho1 / vecOrtho1Abs;
+    const float3_X vecOrtho2 = math::cross(vecOrtho1_norm, vec);
+    vecOrtho1 = vecOrtho1_norm * math::abs(vec);
+
+    return vec * cosTheta +
+           vecOrtho1 * (sinTheta * cosPhi) +
+           vecOrtho2 * (sinTheta * sinPhi);
+}
+
+template<typename T_IonSpecies, typename T_ElectronSpecies, typename T_PhotonSpecies>
+DINLINE
+unsigned int Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::numNewParticles(FrameType& sourceFrame, int localIdx)
+{
+    using namespace PMacc::algorithms;
+
+    PMACC_AUTO(particle, sourceFrame[localIdx]);
+
+    /* particle position, used for field-to-particle interpolation */
+    const floatD_X pos = particle[position_];
+    const int particleCellIdx = particle[localCellIdx_];
+    /* multi-dim coordinate of the local cell inside the super cell */
+    DataSpace<TVec::dim> localCell(DataSpaceOperations<TVec::dim>::template map<TVec > (particleCellIdx));
+    /* interpolation of fieldTmp */
+    const fieldSolver::numericalCellType::traits::FieldPosition<FieldTmp, simDim> fieldTmpPos;
+    const ValueTypeIonCounter ionCount_norm = Field2ParticleInterpolation()
+        (cachedIonCounter.shift(localCell).toCursor(), pos, fieldTmpPos());
+
+    float_X ionDensity = ionCount_norm.x() *
+        (static_cast<float_X>(particleToGrid::derivedAttributes::Counter().getUnit().x()) / CELL_VOLUME);
+
+    const float_X weighting = particle[weighting_];
+
+    const float_X c = SPEED_OF_LIGHT;
+    float3_X mom = particle[momentum_] / weighting;
+    const float_X momAbs = math::abs(mom);
+    float3_X mom_norm = mom / momAbs;
+
+    const float_X mass = frame::getMass<FrameType>();
+    const float_X momAbs_norm = momAbs / (mass * c);
+    const float_X Ekin = (math::sqrt(float_X(1.0) + momAbs_norm*momAbs_norm) - float_X(1.0)) * mass * c*c;
+    if(Ekin < electron::MIN_ENERGY)
+        return 0;
+
+    /* electron deflection */
+    const float_X zMin = float_X(1.0) / (float_X(M_PI)*float_X(M_PI));
+    const float_X zMax = float_X(1.0) / (electron::MIN_THETA*electron::MIN_THETA);
+    const float_X z = zMin + this->randomGen() * (zMax - zMin);
+    const float_X theta = math::rsqrt(z);
+    const float_X theta2 = theta*theta;
+    const float_X rutherfordCoeff = ELECTRON_CHARGE*ELECTRON_CHARGE / (float_X(4.0) * float_X(M_PI) * EPS0) * ion::TARGET_Z / Ekin;
+    const float_X denominator = (electron::MIN_THETA*electron::MIN_THETA + theta2);
+    const float_X scaledDeflectionDCS = float_X(1.0) / float_X(2.0) * (zMax - zMin) * rutherfordCoeff*rutherfordCoeff *
+        float_X(2.0) * float_X(M_PI) * theta2*theta2 / (denominator*denominator);
+    const float_X deflectionProb = ionDensity * c * DELTA_T * scaledDeflectionDCS;
+
+    if(this->randomGen() < deflectionProb)
+    {
+        mom = this->scatterByTheta(mom, theta);
```

Comment:
But it is rotated by theta!

The initial electron momentum has been `mom` (before this line).
The final electron momentum is later `mom + deltaMom * mom_norm`.
The scattered photon momentum is `mom_norm * weighting * photonEnergy / c`.
With `mom_norm = mom / momAbs`. 

So here `mom` (or `mom_norm`) becomes parallel to both the photon momentum and the new electron momentum. I thought the bremsstrahlung as described by F. Salvat et al. just reduces the magnitude of the electron momentum not its direction. 

Is this counter propagation (of photon and electron) here not violating momentum conservation?
```
p_e(before interaction) = mom (before line 170)
p_e(after interaction) = scatterByTheta(mom, theta) + scatterByTheta(mom, theta)/abs(scatterByTheta(mom, theta)) * factor1 = scatterByTheta(mom, theta)/abs(scatterByTheta(mom, theta)) * (1 + factor1) = vec * (1+factor1)
p_ph(after interaction) = scatterByTheta(mom, theta)/abs(scatterByTheta(mom, theta)) * factor2 = vec * factor2
```
so if `scatterByTheta(mom, theta)=vec` is not parallel to `mom (before line 170)`, there is **no**  conservation of momentum. 

What am I assuming wrong here? 
Could you give it a meaningful (`const`) name without reusing old momentum vectors